app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,12 @@
+module Archivable
+  extend ActiveSupport::Concern
+
+  included do
+    scope :archived, -> { where('archived_at is not null') }
+    scope :published, -> { where(archived_at: nil) }
+  end
+
+  def archive
+    update_attributes(archived_at: Time.now)
```

Comment:
I think slightly better is `update_attribute` here.

Also, I'm thinking we don't want to update the time if for some reason we end up trying to archive an already-archived model.

```
update_attribute :archived_at, Time.now unless self.archived_at
```